api/network.go

```diff
@@ -0,0 +1,284 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"net/http"
+
+	"github.com/banzaicloud/pipeline/secret"
+
+	"github.com/banzaicloud/pipeline/auth"
+	"github.com/banzaicloud/pipeline/internal/platform/gin/correlationid"
+	ginutils "github.com/banzaicloud/pipeline/internal/platform/gin/utils"
+	"github.com/banzaicloud/pipeline/internal/providers"
+	"github.com/gin-gonic/gin"
+	"github.com/sirupsen/logrus"
+)
+
+// NetworkAPI implements network functions
+type NetworkAPI struct{}
```

Comment:
Would be nice to inject dependencies here instead of accessing the global versions. Those should eventually go away.

Reply:
Which dependencies do you mean? Could you give an example? I only found the secret store as a global dependency, but that has a "private" type, so it's not trivial to create a reference to it.

Reply:
I mean the logger, errorHandler, db connection dependencies.

Reply:
I added the logger, but couldn't find any more dependencies.